### DISCUSSION TITLE
MacOS işletim sisteminde `strace  -cf ./main` komutu

### DISCUSSION THREAD

**Discussion by Mona-Roza** [gusanmaz/BMB311_OS#8] [Show and tell]
MacOS işletim sisteminde `strace  -cf ./main` komutunun işlevini `dtruss -cf ./main ` komutu yapar. Ancak bu komutu kullanabilmek için öncelikle yapmanız gereken bazı işlemler var. 

1. Öncelikle Apple Menüsünden sistemi kapatın. 
2. Sistemi tekrar başlatırken güç tuşuna bir kere basıp çekmek yerine basılı tutun, böylelikle sistem başlatma seçeneklerine gireceksiniz. -Başlatma seçenekleri yüklenene kadar elinizi güç düğmesinden çekmeyin.-
3. Gelen 2'li seçim ekranında ayar simgesiyle belirtilen seçenekler menüsüne devam edin.
4. Burada şifresini bildiğiniz bir kullanıcı ile devam edin.
5. Öncelikle top bar üzerinde bulunan "İzlenceler" menüsünden "Başlangıç Güvenliği İzlencesi"'ni seçin. 
6. Gelen ekranda güvenlik politikasını ayarlamak için kullanmak istediğiniz sistemi seçin.
7. Disk FileVault ile şifrelenmişse Kilidi Aç’ı tıklayın, parolayı girin, sonra Kilidi Aç’ı tıklayın.
8. Güvenlik Politikası’nı tıklayın.
9. "Azaltılmış Güvenlik" seçeneğini ve bu seçenek aktif edildikten sonra açılan diğer seçenekleri aktive edin.
10. Başlangıç Güvenliği İzlencesini kapatın.
11. Top bar üzerinde bulunan "İzlenceler" menüsünden "Terminal"i seçin.
12. `csrutil disable` komutunu girin. Böylelikle sistem güvenliği kapatılmış olacak. 
13. Bilgisayarınızı normal bir biçimde yeniden başlatın.

Artık terminalde `dtruss -cf ./main` komutunu kullanabilirsiniz. 

**İşlemleriniz bittikten sonra, tekrar Recovery OS modunda terminale girerek `csrutil enable` yapmayı ihmal etmeyin. İhmal durumunda bilgisayar koruması kapalı olacaktır.**

**Comment by yunusemreyazici** [comment]
Teşekkür ederim Başak
